src/main/java/ua/lviv/iot/lab4/expression/Regular.java

```diff
@@ -0,0 +1,46 @@
+package ua.lviv.iot.lab4.expression;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Regular {
```

Comment:
The class name should be as self-descriptive as possible: what is it used for? Imagine a colleague of yours who needs this exact functionality for their task, but since the naming is unclear, they may not find or understand what this class is related to. Then they would:
a. call you and ask wth is that, or 
b. have to read the whole code, or 
c. write a duplicate implementation
And either one is very bad for your project, because a lot of time is wasted.

Reply:
Thank you for your review.